.github/workflows/linter-yaml.yml

```diff
@@ -0,0 +1,93 @@
+name: "[linter] YAML files"
+
+on:
+  pull_request:
+
+env:
+  PYTHONPATH: ${{github.workspace}}
+  PYVER: "3.8"
+  CONFIG_FILES_PATH: "recipes/**/config.yml"
+  CONANDATA_FILES_PATH: "recipes/**/**/conandata.yml"
+
+jobs:
+  test_linter:
+    # A job to run when the linter changes. We want to know in advance how many files will be broken
+    name: Test linter changes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Get changed files
+        uses: tj-actions/changed-files@v20
+        id: changed_files
+        with:
+          files: |
+            linter/**
+
+      - uses: actions/setup-python@v3
+        if: steps.changed_files.outputs.any_changed == 'true'
+        with:
+          python-version: ${{ env.PYVER }}
+      
+      - name: Install dependencies
+        if: steps.changed_files.outputs.any_changed == 'true'
+        run: pip install yamllint
+
+      - name: Run linter (config.yml)
+        if: steps.changed_files.outputs.any_changed == 'true' && always()
+        run: yamllint --config-file linter/yamllint_rules.yml -f parsable ${{ env.CONFIG_FILES_PATH }}
+
+      - name: Run linter (conandata.yml)
+        if: steps.changed_files.outputs.any_changed == 'true' && always()
+        run: yamllint --config-file linter/yamllint_rules.yml -f parsable ${{ env.CONANDATA_FILES_PATH }}
+
+
+  lint_pr_files:
+    # Lint files modified in the pull_request
+    name: Lint YML files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - uses: actions/setup-python@v3
+        with:
+          python-version: ${{ env.PYVER }}
+      
+      - name: Install dependencies
+        run: pip install yamllint
+
+      ## Work on config.yml files
+      - name: Get changed files (config)
+        id: changed_files_config
+        if: always()
+        uses: tj-actions/changed-files@v20
+        with:
+          files: |
+            ${{ env.CONFIG_FILES_PATH }}
+      
+      - name: Run linter (config.yml)
+        if: steps.changed_files_config.outputs.any_changed == 'true' && always()
```

Comment:
why put the `&& always()` condition, when the for loop below won't run anything if no files have been changed ?

Reply:
In this line it is useless, but in L89 it is if we want to run that step regardless of what happened before. Here I added it just for _symmetry_ (in case we reorder the steps and check `conandata` before `config`).

I prefer the action to run the checks on `config.yml` files AND `conandata.yml` files always, IMHO it is better to get all the errors in one execution instead of getting a failure (and stop) from `config.yml` and getting only the errors from `conandata.yml` after fixing the previous ones.